### PR TITLE
[GEP-17] Set owner check configuration in etcd-main Etcd resource

### PR DESCRIPTION
**Pull request by stoyanr**
**How to categorize this PR?**

/area control-plane-migration
/kind enhancement

**What this PR does / why we need it**:
* Enhances the `gardenlet` shoot controller to set the owner check configuration parameters in the `etcd-main` `Etcd` resource. This will cause `etcd-backup-restore` to disable the cluster if the owner domain name no longer resolves to the specified owner ID. These parameters have been introduced with https://github.com/gardener/etcd-druid/pull/239. 
* Introduces a new `spec.settings.ownerChecks` seed setting that enables or disables the creation and checking of owner DNS records for all shoots scheduled on a seed (enabled by default).

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
The `gardenlet` shoot controller will now set the owner check configuration parameters in the `etcd-main` `Etcd` resource. This will cause `etcd-backup-restore` to disable the cluster if the owner domain name no longer resolves to the specified owner ID. The creation and checking of owner DNS record can be disabled via the `spec.settings.ownerChecks` seed setting.
```

### CODE REVIEW

**Comment by stoyanr** [comment]
Rebased on latest master and vendored `etcd-druid` 0.7.0.

**Comment by stoyanr** [comment]
In a new commit, I introduced a new `spec.settings.ownerChecks` seed setting that enables or disables the creation and checking of owner DNS records for all shoots scheduled on a seed (enabled by default).
I consider this PR ready for review now.
/invite @plkokanov @timebertt @timuthy

File: docs/usage/seed_settings.md [hunk 1]
```diff
@@ -59,4 +59,12 @@ In case you want to manage the VPA deployment on your own or have a custom one t
...
+This mechanism relies on the DNS resolution of TXT DNS records being possible and highly reliable, since if the owner check fails the shoot will be effectively disabled for the duration of the failure. 
+In environments where resolving TXT DNS records is either not possible or not considered reliable enough, it may be necessary to disable the owner check mechanism, in order to avoid shoots failing to reconcile or temporary outages due to transient DNS failures. 
+By setting the `.spec.settings.ownerChecks.enabled=false` the creation and checking of owner DNS records can be disabled for all shoots scheduled on this seed. Note that if owner checks are disabled, migrating shoots scheduled on this seed to other seeds should be considered unsafe, and in the future will be disabled as well.
+s
```
Comment (timebertt) [inline docs/usage/seed_settings.md]:
```suggestion
```

File: docs/usage/seed_settings.md [hunk 2]
```diff
@@ -59,4 +59,12 @@ In case you want to manage the VPA deployment on your own or have a custom one t
...
+This mechanism relies on the DNS resolution of TXT DNS records being possible and highly reliable, since if the owner check fails the shoot will be effectively disabled for the duration of the failure. 
+In environments where resolving TXT DNS records is either not possible or not considered reliable enough, it may be necessary to disable the owner check mechanism, in order to avoid shoots failing to reconcile or temporary outages due to transient DNS failures. 
+By setting the `.spec.settings.ownerChecks.enabled=false` the creation and checking of owner DNS records can be disabled for all shoots scheduled on this seed. Note that if owner checks are disabled, migrating shoots scheduled on this seed to other seeds should be considered unsafe, and in the future will be disabled as well.
```
Comment (timebertt) [inline docs/usage/seed_settings.md]:
```suggestion
By setting the `.spec.settings.ownerChecks.enabled=false` (default `true`) the creation and checking of owner DNS records can be disabled for all shoots scheduled on this seed. Note that if owner checks are disabled, migrating shoots scheduled on this seed to other seeds should be considered unsafe, and in the future will be disabled as well.
```

**Review by timebertt — COMMENTED** [review]
Nits only, rest looks good.

File: docs/usage/seed_settings.md (hunk 1 above)
Comment (stoyanr) [reply docs/usage/seed_settings.md]: Done

File: docs/usage/seed_settings.md (hunk 2 above)
Comment (stoyanr) [reply docs/usage/seed_settings.md]: Done

**Comment by stoyanr** [comment]
@timebertt Thanks for your review, addressed the comments in a separate commit.

File: pkg/operation/botanist/etcd.go [hunk 3]
```diff
@@ -108,6 +110,13 @@ func (b *Botanist) DeployEtcd(ctx context.Context) error {
 			Container:            string(secret.Data[v1beta1constants.DataKeyBackupBucketName]),
 			FullSnapshotSchedule: snapshotSchedule,
 		})
+
+		if gardencorev1beta1helper.SeedSettingOwnerChecksEnabled(b.Seed.GetInfo().Spec.Settings) {
```
Comment (plkokanov) [inline pkg/operation/botanist/etcd.go]: Shouldn't we also check for the UseDNSRecords feature gate here? If it is disabled, then the owner DNS record would not exist.

Comment (stoyanr) [reply pkg/operation/botanist/etcd.go]: Yes, absolutely, thanks for noticing. I also thought about double-checking that the feature gate and the seed setting are always checked together (whenever owner checks are concerned) but somehow it slipped my mind.

Comment (stoyanr) [reply pkg/operation/botanist/etcd.go]: Done, PTAL

**Review by timebertt — APPROVED** [review]
Thank you!
/lgtm

**Review by plkokanov — APPROVED** [review]
/lgtm

**Pull request merged**